### ISSUE TITLE
Create a new feature: table widget

### ISSUE THREAD

**Issue by HelmutGoebel** [pojome/elementor#1342]
Tables are needed everywhere, in particular in product presentations, to list features and key numbers. The standard table in the editor is very,very,very bad. Almost no styling options. Please create a table widget with extended styling options. Advanced option could be: Display the content of categories and custom fields (for PRO?).

**Comment by nicolasblume** [comment]
I would love to see some advanced options here like sideways scrolling (with optional fixed scrollbars) for the view on mobile devices.
Another option for mobile devices would be to show the full table first and make it zoomable, so the scrollbars only appear if you zoom in.

the option to exclude some columns or rows on mobile devices would be great as well.

We got some big tables on our website that look really weird on small sceens. 
Another view on mobile devices would be amazing!

**Comment by MasterPayne** [comment]
https://tablepress.org/extensions/responsive-tables/

Have you tried this? It works for me.

**Comment by nicolasblume** [comment]
The tablepress extension looks great, thanks! 

A table widget in elementor would still be nice though

**Comment by Paras16699** [comment]
+1. Tablepress is cool, but I've a requirement where I would need sortable tables on every page which has HTML elements and links in every cell. In Tablepress, we have to write all that HTML by ourselves, and hence, it's a tough task. A responsive table element in Elementor would be great.

**Comment by themedleb** [comment]
Tables is the most waited widget for all Wordpress users since Wordpress doesn't have it natively.

**Comment by mrandt** [comment]
I also vote for this. 

Elementor is great but IMHO a table widget is a must-have requirement. Ever so often we need to present data in a structured way, e.g. to list product technical details, scientific facts or financial figures.

At least it should be possible to define number of rows and columns, add / remove rows or columns and merge cells (spanning).

Preferably I would like to set a row or column to be a header (different styling, perhaps th instead of td elements for screenreaders).

Together with styling per row or cell many use cases could be met.

Scrolling and screen-break-behaviour for repsonsive display would be a nice to have; e.g. sticky rows or columns and scrolling the remaining content separately.

But one thing at a time - a basic widget would already help.

P.S.: I checked several table add-ons to Elementor but was not satisfied with those; I thought those were "oversimplified".

**Comment by ellegaarddk** [comment]
There's a independent plugin out there - haven't tested it much though. 
https://wordpress.org/plugins/tables-for-elementor/

**Comment by mrandt** [comment]
@ellegaarddk I also tested that widget. IMHO it is only geared towards one use case: Comparing two alternatives (e.g. product variants) on high level. Fine for some situations, but unfortunately not what I am looking for.

I would rather like a general table widget that could be adapted to many use cases.

**Comment by themedleb** [comment]
@ellegaarddk I've tested this plugin, it's horrible and doesn't meet all cases of use, I would prefer to use TablePress instead of this one, but if would be so much better to be implemented in Wordpress natively or/and in Elementor officially soon.

**Comment by Strativari** [comment]
I support the request as tables are a very important part of structured data presentation. They often have to be edited by the client so it's important to have a usable and comfortable tool.

**Comment by drank** [comment]
+1
Elementor Extras has something that looks interesting (https://shop.namogo.com/elementor-extras/widgets/table/) but it would be definitely better if this is provided as part of Elementor.

**Comment by ZedProg** [comment]
+1

Elementor team should consider this in high priority. The table is quite a basic feature, which is missing.

**Comment by muazzami** [comment]
+1

Need a Table. I choose Elementor Pro for the client to manage content and there are many tables, i was hoping a simple widget where they can edit cells however there is not.
Please bring it for PRO users at least. Thanks

**Comment by srcek** [comment]
Elementor table element would be awesome. What would be even better is that there's ability to display custom post type content in tables. Like post directory with dynamic link where table columns are mapped to custom fields.

**Comment by HelmutGoebel** [comment]
@srcek Download 'Revolution for Elementor' and use the widget 'Advanced List' and you'll have it already. If you have problems, email me on info@helmutgoebel.de .

**Comment by srcek** [comment]
@HelmutGoebel Thank you for the suggestion and help offer. I'll try that plugin out.

**Comment by MiMoHo** [comment]
Pushing this issue. A widget for tables like Google Sheets with advanced options would be much more convenient to integrate than using the HTML code Google produces for a spreadsheet.

**Comment by ZedProg** [comment]
Hi Team,
Its been a while following this thread, which was  opened on Feb 22, 2017. 
 
May i know when this feature will be available?

I clearly don't understand the delay in propagating such a basic feature. Please add the basic features, before going to the higher one...

**Comment by black-eye** [comment]
+1

**Comment by mylimoproject** [comment]
+1

**Comment by momofone** [comment]
+1

**Comment by catalystmdc** [comment]
+1

**Comment by ArneAmeye** [comment]
+1, i upvoted on the original post up there. Would really be nice to compare data and findings in a structured way. Now it means i have to get out of the Elementor Flow and spend more time on the table than the actual content i'm willing to show.

**Comment by godvsgodvs** [comment]
Yes, this is important. Just ran into a requirement for listing Golf Courses and their fees and features. Now I, also, have to look for a third party plugin that is client friendly.

**Comment by ArekZw** [comment]
+1

**Comment by Turanligil** [comment]
+1

**Comment by sherlanova** [comment]
+1
+1
+1
+1
+1

And I will upvote it hundreds of timesif I could!

This is something that's truly missing from the Pro. I need to use tables for creating client's company profile with detailed information (per section) instead of the narrative style. It's definitely requires tables.

Thank you for the effort and hard work, Elementor team!

**Comment by godvsgodvs** [comment]
sherlanova, It amazes me how Elementor can be so good about so many things but be entirely tone-deaf on others. If you look at the OP you'll see this thread is more than 2 years old and probably isn't monitored anymore.

However, there is a great free plugin for tables that is better than anything they will probably end up with anyway. TablePress is excellent and well maintained. I highly recommend it.

**Comment by sherlanova** [comment]
Aah, I see. TablePress, then? 

It wasn't escaped me the fact that this thread is 2 years old. I thought it was about prioritization or whatever. 

Thank you for the recommendation. Will check it, like, right now.

> sherlanova, It amazes me how Elementor can be so good about so many things but be entirely tone-deaf on others. If you look at the OP you'll see this thread is more than 2 years old and probably isn't monitored anymore.
> 
> However, there is a great free plugin for tables that is better than anything they will probably end up with anyway. TablePress is excellent and well maintained. I highly recommend it.

**Comment by HelmutGoebel** [comment]
Still checking?

**Comment by YujiShen** [comment]
I use Avada and fusion builder allow table, when I try to migrate my content to Elementor I found it is really hard to achieve the same result. Any update on table widget?

**Comment by rodolphebertozzo** [comment]
+1

**Comment by d19dotca** [comment]
Yes, this would be fantastic to add natively to Elementor (perhaps as a Pro widget makes most sense?). I would love to see the table widget be sortable and have the general styling options of other third-party solutions.

**Comment by pradeepdotco** [comment]
Surprised to see this isn't still part of Elementor!

**Comment by 79ho3ein** [comment]
+1

**Comment by paauserrat** [comment]
Hi please, update us on this!

**Comment by paauserrat** [comment]
#16137

**Comment by nicholaszein** [comment]
Hi there,

### 📅  This request is scheduled to be Migrated to the new [Discussions](https://github.com/elementor/elementor/discussions/categories/feature-request) section.

Please do not duplicate this request.

Once we migrate it, we encourage you to cast your **⬆️ Upvote** on the first comment if you feel it should be considered for our roadmap.

**Comment by fabienzensperger** [comment]
We need the table feature since over 6 years. 🤔

**Comment by ayoubkhan558** [comment]
+1
